analyzer/src/main/kotlin/managers/NPM.kt

```diff
@@ -173,7 +175,17 @@ open class NPM(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConf
             val json = it.readValue<ObjectNode>()
             val rawName = json["name"].textValue()
             val (namespace, name) = splitNamespaceAndName(rawName)
-            val version = json["version"].textValue()
+            val version = json["version"].textValue().let { versionFromPackage ->
+                if (versionFromPackage.isSemanticVersion(Semver.SemverType.NPM)) {
+                    versionFromPackage
+                } else {
+                    // This could also be a URL to a tarball or to a Git repository, like e.g.
+                    // https://registry.npmjs.org/form-data/-/form-data-0.2.0.tgz
+                    val fileName = versionFromPackage.substringAfterLast('/')
+                    val versionFromName = fileName.removeSuffix(".tgz").substringAfterLast('-')
+                    versionFromName.takeIf { it.isSemanticVersion(Semver.SemverType.NPM) } ?: versionFromPackage
```

Comment:
Moving this logic to a function would make it unit testable. Approved anyway.

Reply:
Agreed. But I'd like to keep it like this right now for a test scan that I'l be running.